src/NCalc/Expression.cs

```diff
@@ -228,7 +228,9 @@ public bool HasErrors()
         protected Dictionary<string, IEnumerator> ParameterEnumerators;
         protected Dictionary<string, object> ParametersBackup;
 
-        public Func<TResult> ToLambda<TResult>()
+        private struct Void { };
+
+        private Tuple<System.Linq.Expressions.Expression, System.Linq.Expressions.ParameterExpression> ToLinqExpressionInternal<TContext, TResult>()

```

Comment:
This is a personal preference, but can you define a class rather than using this style of Tuple?  I find Item1, Item2, ... aren't the clearest.  

maybe this?
```
public class ContextualExpression
{
  public ParameterExpression Context { get; set; }
  public Expression Expression { get; set; }
}
```

Reply:
Yes I will do, because Tuple is not that convenient as new tuple syntax